pkg/apiserver/profile/router.go

```diff
@@ -0,0 +1,358 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/config"
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/dbstore"
+)
+
+// Built-in component type
+const (
+	tikv = "tikv"
+	tidb = "tidb"
+	pd   = "pd"
+)
+
+// Service is used to provide a kind of feature.
+type Service struct {
+	config *config.Config
+	db     *dbstore.DB
+	tasks  sync.Map
+}
+
+// NewService creates a new service.
+func NewService(config *config.Config, db *dbstore.DB) *Service {
+	err := autoMigrate(db)
+	if err != nil {
+		log.Fatal("Failed to initialize database", zap.Error(err))
+	}
+	return &Service{config: config, db: db, tasks: sync.Map{}}
+}
+
+// Register register the handlers to the service.
+func (s *Service) Register(r *gin.RouterGroup) {
+	endpoint := r.Group("/profile")
+	endpoint.GET("/group/start", s.startHandler)
+	endpoint.GET("/group/status/:groupId", s.statusHandler)
+	endpoint.POST("/group/cancel/:groupId", s.cancelGroupHandler)
+	endpoint.POST("/single/cancel/:taskId", s.cancelHandler)
+	endpoint.GET("/group/download/:groupId", s.downloadGroupHandler)
+	endpoint.GET("/single/download/:taskId", s.downloadHandler)
+	endpoint.DELETE("/group/delete/:groupId", s.deleteHandler)
+}
+
+type ProfilingRequest struct {
+	Tidb []string `form:"tidb" json:"tidb"`
+	Tikv []string `form:"tikv" json:"tikv"`
+	Pd   []string `form:"pd" json:"pd"`
+}
+
+// @Summary Create a task group
+// @Description create and run a task group
+// @Produce json
+// @Param pr query profile.ProfilingRequest true "profiling request"
+// @Success 200 {uint} uint "group ID"
+// @Failure 400 {object} utils.APIError
+// @Router /profile/group/start [post]
+func (s *Service) startHandler(c *gin.Context) {
+	var pr ProfilingRequest
+	if err := c.ShouldBind(&pr); err != nil {
```

Comment:
If you use POST, then this can be ShouldBindJSON and `form:"tidb"` in the field tag can also be removed.